### PR TITLE
ref: add doc for pessimistic auto-commit

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Jack Yu <jackysp@gmail.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Add doc for new option: `pessimistic-auto-commit` ref: https://github.com/pingcap/tidb/pull/32056

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.0 (TiDB 5.4 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by MyonKeminta — APPROVED** [review]

File: tidb-configuration-file.md [hunk 1]
```diff
@@ -706,6 +706,15 @@ TiDB 服务状态相关配置。
 + 控制 [`INFORMATION_SCHEMA.DEADLOCKS`](/information-schema/information-schema-deadlocks.md) 表中是否收集可重试的死锁错误信息。详见 `DEADLOCKS` 表文档的[可重试的死锁错误](/information-schema/information-schema-deadlocks.md#可重试的死锁错误)小节。
 + 默认值：false
 
+### pessimistic-auto-commit
+
++ 用来控制开始全局悲观事务模式下（`tidb_txn_mode='pessimistic'`)时，自动提交的事务使用的事务模式。默认情况下，即使开始全局悲观事务模式，自动提交事务依然使用乐观事务方式来执行。
```
Comment (sticnarf) [inline tidb-configuration-file.md:711]:
```suggestion
+ 用来控制开启全局悲观事务模式下（`tidb_txn_mode='pessimistic'`)时，自动提交的事务使用的事务模式。默认情况下，即使开启全局悲观事务模式，自动提交事务依然使用乐观事务方式来执行。
```

**Comment by jackysp** [comment]
PTAL @sticnarf , thx!

**Review by sticnarf — APPROVED** [review]

**Comment by ran-huang** [comment]
@jackysp Does this apply to v6.0?

**Comment by jackysp** [comment]
> @jackysp Does this apply to v6.0?

Yes.

File: tidb-configuration-file.md [hunk 2]
```diff
@@ -706,6 +706,15 @@ TiDB 服务状态相关配置。
 + 控制 [`INFORMATION_SCHEMA.DEADLOCKS`](/information-schema/information-schema-deadlocks.md) 表中是否收集可重试的死锁错误信息。详见 `DEADLOCKS` 表文档的[可重试的死锁错误](/information-schema/information-schema-deadlocks.md#可重试的死锁错误)小节。
 + 默认值：false
 
+### pessimistic-auto-commit
+
++ 用来控制开启全局悲观事务模式下（`tidb_txn_mode='pessimistic'`)时，自动提交的事务使用的事务模式。默认情况下，即使开启全局悲观事务模式，自动提交事务依然使用乐观事务方式来执行。
+当开启本开关后（设置为 true），在全局悲观事务模式下，自动提交事务将也使用悲观事务模式执行。行为与其他显示提交的悲观事务相同。
+对于存在冲突的场景，开启本开关可以将自动提交事务纳入全局等锁管理中，从而避免死锁，改善冲突造成死锁带来的业务尖刺。
+对于不存在冲突的场景，如果有大量自动提交事务且单个事务操作数据量较大的情况下，开启本开关会造成性能回退，如，自动提交的`insert into select`语句。
```
Comment (ran-huang) [inline tidb-configuration-file.md:714]:
```suggestion
+ 用来控制开启全局悲观事务模式下 (`tidb_txn_mode='pessimistic'`) 时，自动提交的事务使用的事务模式。默认情况下，即使开启全局悲观事务模式，自动提交事务依然使用乐观事务模式来执行。当开启该配置项后（设置为 `true`），在全局悲观事务模式下，自动提交事务将也使用悲观事务模式执行。行为与其他显式提交的悲观事务相同。
+ 对于存在冲突的场景，开启本开关可以将自动提交事务纳入全局等锁管理中，从而避免死锁，改善冲突造成死锁带来的业务尖刺。
+ 对于不存在冲突的场景，如果有大量自动提交事务且单个事务操作数据量较大的情况下，开启该配置项会造成性能回退。例如，自动提交的 `INSERT INTO SELECT` 语句。
```

Comment (ran-huang) [reply tidb-configuration-file.md:714]: 不推荐使用行话黑话、俚语等，“业务尖刺“能否换一种更通用的表达？

Comment (jackysp) [reply tidb-configuration-file.md]: 这里其实应该说时延尖刺，或者说业务上的时延尖刺，英文的说法应该是 latency Spike

Comment (jackysp) [reply tidb-configuration-file.md]: 换成时延尖刺？响应尖刺？

Comment (ran-huang) [reply tidb-configuration-file.md]: 感谢解答，用“时延尖刺”吧

Comment (jackysp) [reply tidb-configuration-file.md]: Done.

File: tidb-configuration-file.md [hunk 3]
```diff
@@ -706,6 +706,14 @@ TiDB 服务状态相关配置。
 + 控制 [`INFORMATION_SCHEMA.DEADLOCKS`](/information-schema/information-schema-deadlocks.md) 表中是否收集可重试的死锁错误信息。详见 `DEADLOCKS` 表文档的[可重试的死锁错误](/information-schema/information-schema-deadlocks.md#可重试的死锁错误)小节。
 + 默认值：false
 
+### pessimistic-auto-commit
+
++ 用来控制开启全局悲观事务模式下 (`tidb_txn_mode='pessimistic'`) 时，自动提交的事务使用的事务模式。默认情况下，即使开启全局悲观事务模式，自动提交事务依然使用乐观事务模式来执行。当开启该配置项后（设置为 `true`），在全局悲观事务模式下，自动提交事务将也使用悲观事务模式执行。行为与其他显式提交的悲观事务相同。
++ 对于存在冲突的场景，开启本开关可以将自动提交事务纳入全局等锁管理中，从而避免死锁，改善冲突造成死锁带来的时延尖刺。
++ 对于不存在冲突的场景，如果有大量自动提交事务且单个事务操作数据量较大的情况下，开启该配置项会造成性能回退。例如，自动提交的 `INSERT INTO SELECT` 语句。
+
```
Comment (TomShawn) [inline tidb-configuration-file.md:714]:
```suggestion
```

File: tidb-configuration-file.md [hunk 4]
```diff
@@ -706,6 +706,14 @@ TiDB 服务状态相关配置。
 + 控制 [`INFORMATION_SCHEMA.DEADLOCKS`](/information-schema/information-schema-deadlocks.md) 表中是否收集可重试的死锁错误信息。详见 `DEADLOCKS` 表文档的[可重试的死锁错误](/information-schema/information-schema-deadlocks.md#可重试的死锁错误)小节。
 + 默认值：false
 
+### pessimistic-auto-commit
```
Comment (TomShawn) [inline tidb-configuration-file.md:709]:
```suggestion
### pessimistic-auto-commit（在 v6.0 版本开始引入）
```

**Review by TomShawn — APPROVED** [review]
Rest LGTM

File: tidb-configuration-file.md [hunk 5]
```diff
@@ -706,6 +706,14 @@ TiDB 服务状态相关配置。
...
+
++ 用来控制开启全局悲观事务模式下 (`tidb_txn_mode='pessimistic'`) 时，自动提交的事务使用的事务模式。默认情况下，即使开启全局悲观事务模式，自动提交事务依然使用乐观事务模式来执行。当开启该配置项后（设置为 `true`），在全局悲观事务模式下，自动提交事务将也使用悲观事务模式执行。行为与其他显式提交的悲观事务相同。
++ 对于存在冲突的场景，开启本开关可以将自动提交事务纳入全局等锁管理中，从而避免死锁，改善冲突造成死锁带来的时延尖刺。
```
Comment (TomShawn) [inline tidb-configuration-file.md:712]: “时延尖刺”是指较高的延迟吗？

Comment (jackysp) [reply tidb-configuration-file.md:712]: 是

Comment (TomShawn) [reply tidb-configuration-file.md:712]: 能否用更通俗易懂的表达呢？
```suggestion
+ 对于存在冲突的场景，开启本开关可以将自动提交事务纳入全局等锁管理中，从而避免死锁，改善冲突造成死锁带来较高的延迟。
```

Comment (jackysp) [reply tidb-configuration-file.md:712]: 不，这样展示不出来突发性的高

Comment (jackysp) [reply tidb-configuration-file.md:712]: 参见 https://github.com/pingcap/docs-cn/pull/8396#discussion_r813520421

File: tidb-configuration-file.md (hunk 4 above)
Comment (en-jin19) [reply tidb-configuration-file.md:709]: > 
```suggestion
### pessimistic-auto-commit（从 v6.0.0 版本开始引入）
```

File: tidb-configuration-file.md [hunk 6]
```diff
@@ -706,6 +706,14 @@ TiDB 服务状态相关配置。
 + 控制 [`INFORMATION_SCHEMA.DEADLOCKS`](/information-schema/information-schema-deadlocks.md) 表中是否收集可重试的死锁错误信息。详见 `DEADLOCKS` 表文档的[可重试的死锁错误](/information-schema/information-schema-deadlocks.md#可重试的死锁错误)小节。
 + 默认值：false
 
+### pessimistic-auto-commit
+
++ 用来控制开启全局悲观事务模式下 (`tidb_txn_mode='pessimistic'`) 时，自动提交的事务使用的事务模式。默认情况下，即使开启全局悲观事务模式，自动提交事务依然使用乐观事务模式来执行。当开启该配置项后（设置为 `true`），在全局悲观事务模式下，自动提交事务将也使用悲观事务模式执行。行为与其他显式提交的悲观事务相同。
```
Comment (en-jin19) [inline tidb-configuration-file.md:711]: 用来控制开启全局悲观事务模式 (`tidb_txn_mode='pessimistic'`) 时，

**Pull request merged**